docs/get-started/support-new-model.md

```diff
@@ -0,0 +1,548 @@
+truediv = matmul / 8.0;  matmul = None
+add_1 = truediv + mul;  truediv = None
+softmax = torch.nn.functional.softmax(add_1, dim = -1);  add_1 = None
+matmul_1 = torch.matmul(softmax, permute_1);  softmax = permute_1 = None
+
+```
+
+We now have our pattern to catch in the model, to make the pattern easier to read, we rename the following nodes:
+
+- `permute` → `k`
+- `permute_1` → `v`
+- `permute_2` → `q`
+- `mul` → `attention_mask`
+
+and can write our pattern function:
+
+``` { .py }
+def pattern(q, k, attention_mask, v):
+    transpose = k.transpose(-1, -2)
+    matmul = torch.matmul(q, transpose)
+    truediv = matmul / 8.0
+    add_1 = truediv + attention_mask
+    softmax = torch.nn.functional.softmax(add_1, dim=-1)
+    matmul_1 = torch.matmul(softmax, v)
+    return matmul_1
+```
+
+### Replace the Attention part
+
+We now needs to add our replace function to call the optimized kernel. We can see in [kernl/model_optimization.py](https://github.com/ELS-RD/kernl/blob/9c3ec9b3c03609c20ca9a00eda7a58d4769bf47c/src/kernl/implementations/attention.py#L492) the optimized attention kernel needs in addition to `q`, `k`, `v` and `attention_mask`, the `output` and the `sm_scale` parameter.
+
+``` { .py }
+def attention_forward(
+    q: torch.Tensor,
+    k: torch.Tensor,
+    v: torch.Tensor,
+    output: torch.Tensor,
+    sm_scale: float,
+    is_causal: bool = False,
+    attention_mask: Optional[torch.Tensor] = None,
+):
+```
+
+The `output` parameter is simply the resulting tensor. We need to provide the tensor beforehand.
+
+The `sm_scale` parameter corresponds to the scale factor applied to the query-key compatibility function in the attention function. Defined as $\frac{1}{\sqrt{d_k}}$, it correspond to the `true_div` node in the FX graph. In this case `sm_scale` is $\frac{1}{8.0}$.
+
+We can now write our replacement part by calling the optimized kernel:
+
+``` { .py }
+torch.fx.wrap("attention_forward")
+
+def replace(q, k, attention_mask, v):
+    output = torch.empty_like(q)
+    output = attention_forward(q, k, v, output, 1 / 8.0, is_causal=False, attention_mask=attention_mask)
+    return output
+```
+
+To wrap it up, we can define our replacement function:
+
+``` { .py }
+import torch
+
+from kernl.implementations.attention import attention_forward
+from kernl.utils.extended_matcher import replace_pattern
+
+
+torch.fx.wrap("attention_forward")
+
+def replace_attention(gm: torch.fx.GraphModule):
+
+    def pattern(q, k, attention_mask, v):
+        transpose = k.transpose(-1, -2)
+        matmul = torch.matmul(q, transpose)
+        truediv = matmul / 8.0
+        add_1 = truediv + attention_mask
+        softmax = torch.nn.functional.softmax(add_1, dim=-1)
+        matmul_1 = torch.matmul(softmax, v)
+        return matmul_1
+
+    def replace(q, k, attention_mask, v):
+        output = torch.empty_like(q)
+        output = attention_forward(q, k, v, output, 1 / 8.0, is_causal=False, attention_mask=attention_mask)
```

Comment:
maybe explain what is `is_causal` ? maybe in another doc, I don't know

Reply:
As it's already a bit complicated, I've explained the minimal way to wrap the attention kernel. But, if it's misleading I can add a few words about it.